packages/redux-genes/src/GenePageHoc.js

```diff
@@ -1,85 +1,139 @@
-/* eslint-disable react/no-unused-prop-types */
-/* eslint-disable no-shadow */
-/* eslint-disable import/no-unresolved */
-/* eslint-disable import/extensions */
-
-import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
 import { actions as variantActions } from '@broad/redux-variants'
-import { actions as tableActions } from '@broad/table'
-import { currentGene, currentTranscript, isFetching, geneData, geneNotFound, actions as geneActions } from './index'
+import { Loading } from '@broad/ui'
+
+import { actions as geneActions } from './genes'
+
 
 const GenePageContainer = ComposedComponent => class GenePage extends Component {
   static propTypes = {
-    currentGene: PropTypes.string,
-    gene: PropTypes.object,
-    isFetching: PropTypes.bool.isRequired,
-    fetchGeneIfNeeded: PropTypes.func.isRequired,
-    resetSearchVariants: PropTypes.func.isRequired,
+    geneName: PropTypes.string.isRequired,
+    loadGeneData: PropTypes.func.isRequired,
+    transcriptId: PropTypes.string,
   }
 
   static defaultProps = {
-    gene: null,
-    currentGene: null,
+    transcriptId: undefined,
+  }
+
+  state = {
+    geneData: null,
+    isLoading: false,
+    loadError: null,
   }
 
   componentDidMount() {
-    const { currentGene, match, fetchGeneIfNeeded } = this.props
-    fetchGeneIfNeeded(currentGene, match, history)
+    this.mounted = true
+    this.loadGeneData()
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { fetchGeneIfNeeded, currentGene, currentTranscript, history } = this.props
-    if (currentGene !== nextProps.currentGene ||
-      currentTranscript !== nextProps.currentTranscript
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.geneName !== prevProps.geneName
+      || this.props.transcriptId !== prevProps.transcriptId
     ) {
-      // if(this.props.route.path == nextProps.route.path) return false
-      history.push(`/gene/${nextProps.currentGene}`)
-      fetchGeneIfNeeded(nextProps.currentGene)
-      this.props.resetSearchVariants()
-      this.props.resetFilter()
+      this.loadGeneData()
     }
   }
 
+  componentWillUnmount() {
+    this.mounted = false
+  }
+
+  loadGeneData() {
```

Comment:
Slightly confusing to have two functions named this in the same component

Reply:
Agreed. A more appropriate name for the function passed in would be `fetchGeneData`, though I wavered on that since it has a side effect of changing the store. But I suppose the GenePage shouldn't care or be aware of that. Will change.